types/address.go

```diff
@@ -239,37 +235,33 @@ func (aa AccAddress) Bytes() []byte {
 
 // AccAddress.String() is expensive and if unoptimized dominantly showed up in profiles,
 // yet has no mechanisms to trivially cache the result given that AccAddress is a []byte type.
-// With the string interning below, we are able to achieve zero cost allocations for string->[]byte
-// because the Go compiler recognizes the special case of map[string([]byte)] when used exactly
-// in that pattern. See https://github.com/cosmos/cosmos-sdk/issues/8693.
 var addMu sync.Mutex
-var addrStrMemoize = make(map[string]string)
+var addrStrMemoize, _ = simplelru.NewLRU(1000, nil)
```

Comment:
1000 seems fairly small here. If we bump it to 60k (kind of reasonable) the memory footprint is 19.2 megabytes at with a size 32 bytes per address. I would feel comfortable with this size.

Reply:
the storage is address size + key size + hasha map residual space. I was thinking about number of address can fit in on block. Sure we can bump to bigger size. So how about 30k , to take into account the `key` storage?

Reply:
Updated to 30k